### DISCUSSION TITLE
$queryRaw add integer list to IN clause

### DISCUSSION THREAD

**Discussion by ddowningms** [prisma/prisma#12917] [Q&A] [answered]
I'm running Prisma on a Windows 10 computer and using Visual Studio Code to develop my API.  I'm developing in JavaScript.

I'm trying to use the $queryRaw and include a list of integers with an 'IN' clause on the query.

  `... IN (${intList.join().split(',').map((i) => `${i}`).join()})...`
  `... IN (${intList.join()})...`

Everything I've tried fails, an example of one of the errors is:

- Raw query failed. Code: `245`. Message: `Conversion failed when converting the nvarchar value '527830,527963,528408,527835,527956,527839' to data type int.`

If I hardcode the list in the list in the query, it works, but if I generate a string and use the $queryRaw method as follows:

prisma.$queryRaw`${generatedString}`

it fails with errors like <name> is not a valid identifier... is it possible to use a string that was constructed outside the $queryRaw string?

const queryString = ``SELECT .......  WHERE Col IN (${intArray.join()}) ...``

var response = await prisma.$queryRaw`${queryString}`

Any help is greatly appreciated.

**Comment by ddowningms** [comment, marked as answer]
After countless attempts, I was able to get it working by changing the require statement from:

const { PrismaClient } = require('@prisma/client');

to

const PrismaClient = require('@prisma/client');

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
